rust/macros/module.rs

```diff
@@ -300,6 +300,25 @@ impl ModuleInfo {
 
         info
     }
+
+    /// Parse ["First", "Second"] into Some(vec!["First", "Second"])
+    fn parse_authors(it: &mut token_stream::IntoIter) -> Vec<String> {
+        let mut authors: Vec<String> = vec![];
+        let group = expect_group(it);
+        assert_eq!(group.delimiter(), Delimiter::Bracket);
+        let mut stream = group.stream().into_iter();
+        loop {
+            let author = expect_string(&mut stream);
+            authors.push(author);
+            if let Some(punct) = try_punct(&mut stream) {
+                assert_eq!(punct, ',');
+            } else {
+                break;
```

Comment:
I think this will currently parse a stream where `try_punct` fails because the next token isn't a punct. E.g. `["A", "B" "C"]` will successfully parse as `vec!["A", "B"]` because the missing comma before `"C"` breaks us out of the loop. We probably need to check that there isn't anything extra left in the stream too.